docs/configuration/protocols/segment-routing.rst

```diff
@@ -0,0 +1,564 @@
+different portions of the network path that the packet will take.
+
+The SR segments are portions of the network path taken by the packet, and are 
+called SIDs. At each node, the first SID of the list is read, executed as a 
+forwarding function, and may be popped to let the next node read the next SID of 
+the list. The SID list completely determines the path where the packet is 
+forwarded.
+
+Segment Routing can be applied to an existing MPLS-based data plane and defines
+a control plane network architecture. In MPLS networks, segments are encoded as
+MPLS labels and are added at the ingress router. These MPLS labels are then 
+exchanged and populated by Interior Gateway Protocols (IGPs) like IS-IS or OSPF 
+which are running on most ISPs.
+
+
+.. note:: Segment routing defines a control plane network architecture and
+  can be applied to an existing MPLS based dataplane. In the MPLS networks,
+  segments are encoded as MPLS labels and are imposed at the ingress router.
+  MPLS labels are exchanged and populated by IGPs like IS-IS.Segment Routing
+  as per RFC8667 for MPLS dataplane. It supports IPv4, IPv6 and ECMP and has
+  been tested against Cisco & Juniper routers.however,this deployment is still
+  EXPERIMENTAL for FRR .
+ 
+
+IS-IS SR Configuration
+----------------------
+
+Segment routing (SR) is used by the IGP protocols to interconnect network
+devices, below configuration shows how to enable SR on IS-IS:
+
+
+.. note:: ``Known limitations:`` 
+
+  No support for level redistribution (L1 to L2 or L2 to L1)
+
+  No support for binding SID
+
+  No support for SRLB
+
+  Only one SRGB and default SPF Algorithm is supported
+
+
+
+.. cfgcmd:: set protocols isis segment-routing global-block <high-label-value 
+  | low-label-value> <label-value>
+
+  Set the Segment Routing Global Block i.e. the label range used by MPLS to 
+  store label in the MPLS FIB for Prefix SID. Note that the block size may 
+  not exceed 65535.
+
+.. cfgcmd:: set protocols isis segment-routing local-block <high-label-value 
+  | low-label-value> <label-value>
+
+  Set the Segment Routing Local Block i.e. the label range used by MPLS to 
+  store label in the MPLS FIB for Prefix SID. Note that the block size may 
+  not exceed 65535.Segment Routing Local Block, The negative command always 
+  unsets both.
+
+.. cfgcmd:: set protocols isis segment-routing maximum-label-depth <1-16>
+
+  Set the Maximum Stack Depth supported by the router. The value depend of
+  the MPLS dataplane.
+
+.. cfgcmd:: set protocols isis segment-routing prefix <address> index value 
+  <0-65535>
+   
+  A segment ID that contains an IP address prefix calculated by an IGP in the
+  service provider core network. Prefix SIDs are globally unique, this value
+  indentify it 
+
+.. cfgcmd:: set protocols isis segment-routing prefix <address> index
+   <no-php-flag | explicit-null| n-flag-clear>
+
+   prefix. The ‘no-php-flag’ means NO Penultimate Hop Popping that allows SR 
```

Comment:
i think the `prefix.` belong the above block?

Reply:
done